tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/etc/ToEtcExpr.scala

```diff
@@ -152,6 +150,22 @@ class ToEtcExpr(annotationStore: AnnotationStore, varPool: TypeVarPool) extends
    * @return an expression in the simply typed lambda calculus varient Etc
    */
   def apply(ex: TlaEx): EtcExpr = {
+    val tex = transform(ex)
+
+    ex.typeTag match {
+      case Typed(typeInTag: TlaType1) =>
+        // the expression has a type tag, so we use an identity operator
+        val a = varPool.fresh
+        val identity = OperT1(Seq(a, a), a)
+        val blameRef = BlameRef(ex.ID)
+        mkApp(blameRef, Seq(identity), mkConst(blameRef, typeInTag), tex)
```

Comment:
I don't understand why we introduce an identity operator here. Should you perhaps add a line or two in the comments  to motivate this?

Reply:
Good point. I am now receiving complaints by the type checker about this. It is quite confusing. Basically, I want to enforce that the expression has the same type as the one in the tag.

Reply:
I have fixed that in #641, as it was popping up there too